lib/Doctrine/Common/Cache/CouchbaseBucketCache.php

```diff
@@ -0,0 +1,241 @@
+            ]);
+        } catch (Exception $e) {
+            // TODO Log exception.
+            return false;
+        }
+
+        if ($document instanceof Document) {
+            return !$document->error;
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doDelete($id)
+    {
+        $id = $this->normalizeKey($id);
+
+        try {
+            $document = $this->bucket->remove($id);
+        } catch (Exception $e) {
+            if ($e->getCode() === 13) {
+                return true;
+            }
+
+            // TODO Log exception.
+            return false;
+        }
+
+        if ($document instanceof Document) {
+            return !$document->error;
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doFlush()
+    {
+        $manager = $this->bucket->manager();
+
+        // Flush does not return with success or failure, and must be enabled per bucket on the server.
+        // Store a marker item so that we will know if it was successful.
+        $this->doSave(__METHOD__, true, 60);
+
+        $manager->flush();
+
+        if ($this->doContains(__METHOD__)) {
+            $this->doDelete(__METHOD__);
+
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function doGetStats()
+    {
+        $manager = $this->bucket->manager();
+        $stats   = $manager->info();
+        $nodes = $stats['nodes'];
+        $node = $nodes[0]; // TODO Support more than 1 node.
+        $interestingStats = $node['interestingStats'];
+
+        return [
+            Cache::STATS_HITS   => $interestingStats['get_hits'],
+            Cache::STATS_MISSES => $interestingStats['cmd_get'] - $interestingStats['get_hits'],
+            Cache::STATS_UPTIME => $node['uptime'],
+            Cache::STATS_MEMORY_USAGE     => $interestingStats['mem_used'],
+            Cache::STATS_MEMORY_AVAILABLE => $node['memoryFree'],
+        ];
+    }
+
+    /**
+     * @param $id
+     * @return bool|string
+     */
+    private function normalizeKey($id)
+    {
+        $normalized = substr($id, 0, 250);
+
+        if ($normalized === false) {
+            // TODO Log it?
+            return $id;
+        }
+
+        return $normalized;
+    }
+
+    /**
+     * @param mixed $value
+     * @return mixed
+     */
+    private function encode($value)
+    {
+        if (is_array($value) || is_object($value) || is_numeric($value)) {
+            return serialize($value);
+        }
+
+        if (is_string($value) && $value) {
+            return $value;
+        }
+
+        return json_encode($value);
+    }
+
+    /**
+     * @param mixed $value
+     * @return mixed
+     */
+    private function decode($value)
+    {
+        $decoded = @unserialize($value);
+
+        if ($decoded !== false) {
+            return $decoded;
+        }
+
+        $decoded = json_decode($value);
```

Comment:
how embarrassing. thank you for calling this out. i had some other bugs in the class earlier that evening which were resolved by mixing formats. after those were resolved this blend was no longer necessary.